app/Helpers/GeometryHelper.php

```diff
@@ -33,4 +35,85 @@ public static function getPolygonsBbox($polygonsIds)
 
         return $bboxCoordinates;
     }
+
+    public function updateProjectCentroid(string $projectUuid)
+    {
+        try {
+            $centroid = $this->centroidOfProject($projectUuid);
+
+            if ($centroid === null) {
+                Log::warning("Invalid centroid for projectUuid: $projectUuid");
+            }
+
+            $centroidArray = json_decode($centroid, true);
+
+            $latitude = $centroidArray['coordinates'][1];
+            $longitude = $centroidArray['coordinates'][0];
+
+
+            Project::where('uuid', $projectUuid)
+                ->update([
+                    'lat' => $latitude,
+                    'long' => $longitude,
+                ]);
+
+
+            Log::info("Centroid updated for projectUuid: $projectUuid");
+
+            return 'Centroids updated successfully!';
+        } catch (\Exception $e) {
+            Log::error("Error updating centroid for projectUuid: $projectUuid");
+
+            return response()->json([
+              'message' => 'Error updating centroid',
+              'error' => $e->getMessage(),
+            ], 500);
+        }
+
+    }
+
+    public function centroidOfProject($projectUuid)
+    {
+        $project = Project::where('uuid', $projectUuid)->first();
+
+        if (! $project) {
+            return null;
+        }
+
+        $sitePolygons = $project->sitePolygons;
+
+        if ($sitePolygons->isEmpty()) {
+            return null; // Return null if no polygons are found for the given projectUuid
+        }
+
+        $polyIds = $sitePolygons->pluck('poly_id')->toArray();
```

Comment:
I left this feedback on the other PR as well (is there duplicate code in these PRs? In the future, please base a second PR on the branch of the first so we can avoid reviewing code twice). This would be more efficient like so:
```
$polyIds = $project->sitePolygons()->pluck('poly_id')->toArray();
if (empty($polyIds)) {
  return null; // Return null if no polygons are found for the given projectUuid
}
```

Reply:
Done.